File: board-admin-backend/src/main/java/com/portSrilanka/board_admin_backend/dto/agenda/AgendaItemRequest.java
package com.portSrilanka.board_admin_backend.dto.agenda;

import com.portSrilanka.board_admin_backend.enums.AgendaItemType;
import lombok.Data;

@Data
public class AgendaItemRequest {
    private Long meetingId;
    private Long sectionId;
    private AgendaItemType itemType;
    private String title;
    private String numberLabel;
    private Integer displayOrder;
    private String description;
    private String mediaPath;
}

File: board-admin-frontend/frontend/lib/features/agendas/model/agenda_item_request.dart
class AgendaItemRequest {
  final int meetingId;
  final int? sectionId;
  final String itemType;
  final String title;
  final String? numberLabel;
  final int? displayOrder;
  final String? description;
  final String? mediaPath;

  AgendaItemRequest({
    required this.meetingId,
    this.sectionId,
    required this.itemType,
    required this.title,
    this.numberLabel,
    this.displayOrder,
    this.description,
    this.mediaPath,
  });

  Map<String, dynamic> toJson() {
    return {
      'meetingId': meetingId,
      'sectionId': sectionId,
      'itemType': itemType,
      'title': title,
      'numberLabel': numberLabel,
      'displayOrder': displayOrder,
      'description': description,
      'mediaPath': mediaPath,
    };
  }
}
